### PR TITLE
add path to API only for HTTP operations

### PR DESCRIPTION
**Pull request by panpann**
hi,

First thing, thank you for creating and maintaining connexion !

this pull request allows to use [all path item fields](https://swagger.io/specification/#pathItemObject).

I encountered an error while creating an extension using **x-** vars in path object.  
The code was skipping only the field **Parameters** and tried to create an operation on others fields.

### stacktrace
```
    def test_using_all_fields_in_path_item(simple_api_spec_dir):
        app = App(__name__, specification_dir=simple_api_spec_dir)
>       app.add_api('openapi.yaml')

tests/api/test_bootstrap.py:243: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../.local/share/virtualenvs/connexion-LpTILqQi/lib/python3.6/site-packages/connexion/apps/flask_app.py:54: in add_api
    api = super(FlaskApp, self).add_api(specification, **kwargs)
../../.local/share/virtualenvs/connexion-LpTILqQi/lib/python3.6/site-packages/connexion/apps/abstract.py:155: in add_api
    options=api_options.as_dict())
../../.local/share/virtualenvs/connexion-LpTILqQi/lib/python3.6/site-packages/connexion/apis/abstract.py:108: in __init__
    self.add_paths()
../../.local/share/virtualenvs/connexion-LpTILqQi/lib/python3.6/site-packages/connexion/apis/abstract.py:217: in add_paths
    self._handle_add_operation_error(path, method, sys.exc_info())
../../.local/share/virtualenvs/connexion-LpTILqQi/lib/python3.6/site-packages/connexion/apis/abstract.py:228: in _handle_add_operation_error
    six.reraise(*exc_info)
../../.local/share/virtualenvs/connexion-LpTILqQi/lib/python3.6/site-packages/six.py:693: in reraise
    raise value
../../.local/share/virtualenvs/connexion-LpTILqQi/lib/python3.6/site-packages/connexion/apis/abstract.py:207: in add_paths
    self.add_operation(path, method)
../../.local/share/virtualenvs/connexion-LpTILqQi/lib/python3.6/site-packages/connexion/apis/abstract.py:170: in add_operation
    pass_context_arg_name=self.pass_context_arg_name
../../.local/share/virtualenvs/connexion-LpTILqQi/lib/python3.6/site-packages/connexion/operations/__init__.py:8: in make_operation
    return spec.operation_cls.from_spec(spec, *args, **kwargs)
../../.local/share/virtualenvs/connexion-LpTILqQi/lib/python3.6/site-packages/connexion/operations/openapi.py:138: in from_spec
    **kwargs
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <connexion.operations.openapi.OpenAPIOperation object at 0x7fa49c011860>
api = <connexion.apis.flask_api.FlaskApi object at 0x7fa49c4c2ac8>
method = 'summary', path = '/add_operation_on_http_methods_only'
operation = 'this is a test'
resolver = <connexion.resolver.Resolver object at 0x7fa49c0ad780>
path_parameters = [{'description': 'Name of the person to greet.', 'in': 'path', 'name': 'name', 'required': False, ...}]
app_security = None
components = {'examples': {'justAnExample': {'summary': 'a basic example.', 'value': 'Good evening, doctor.'}}, 'parameters': {'Opt...'description': 'Docker image version to deploy', 'type': 'string'}}, 'required': ['image_version'], 'type': 'object'}}}
validate_responses = False, strict_validation = False, randomize_endpoint = None
validator_map = None, pythonic_params = False, uri_parser_class = None
pass_context_arg_name = None

    def __init__(self, api, method, path, operation, resolver, path_parameters=None,
                 app_security=None, components=None, validate_responses=False,
                 strict_validation=False, randomize_endpoint=None, validator_map=None,
                 pythonic_params=False, uri_parser_class=None, pass_context_arg_name=None):
        """
        This class uses the OperationID identify the module and function that will handle the operation
    
        ...
        """
        self.components = components or {}
    
        def component_get(oas3_name):
            return self.components.get(oas3_name, {})
    
        # operation overrides globals
        security_schemes = component_get('securitySchemes')
>       app_security = operation.get('security', app_security)
E       AttributeError: 'str' object has no attribute 'get'

../../.local/share/virtualenvs/connexion-LpTILqQi/lib/python3.6/site-packages/connexion/operations/openapi.py:70: AttributeError
```

### CODE REVIEW

File: connexion/apis/abstract.py [hunk 1]
```diff
@@ -199,7 +200,7 @@ def add_paths(self, paths=None):
             logger.debug('Adding %s%s...', self.base_path, path)
 
             for method in methods:
-                if method == 'parameters':
```
[the remark this answers is not in the document]
Comment (panpann) [reply connexion/apis/abstract.py]: Yes I made it case sensitive, I read the openapi spec which says `All field names in the specification are case sensitive. This includes all fields that are used as keys in a map, except where explicitly noted that keys are case insensitive.`

You want it insensitive ?

Also, I ran tests with the case insensitive test on my laptop, and have always the same errors. Do you have a hint about thoses errors ?

[the remark this answers is not in the document]
Comment (dtkav) [reply connexion/apis/abstract.py]: nah, you're right. That's what I get for reading diffs on my phone :)

File: connexion/http_facts.py [hunk 2]
```diff
@@ -2,3 +2,14 @@
     'application/x-www-form-urlencoded',
     'multipart/form-data'
 ]
+
+METHODS = set([
+    "get",
+    "put"
```
Comment (dtkav) [inline connexion/http_facts.py]: It turns out the issue is a missing comma here.

**Comment by dtkav** [comment]
nice work - just fix that comma and you should be good to go

File: connexion/http_facts.py (hunk 2 above)
Comment (panpann) [reply connexion/http_facts.py]: damn... sorry for that aha

I will add a check in my test, this evening, to see if a route has been created in flask app for each method, before merging

**Comment by panpann** [comment]
Here is a test to cover the case we got with the missing coma.

**Comment by jmcs** [comment]
:+1:

**Pull request merged**